internal/service/mongo.go

```diff
@@ -0,0 +1,83 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/etf1/kafka-mongo-watcher/internal/mongo"
+	"github.com/gol4ng/logger"
+	mongodriver "go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func (container *Container) GetMongoReplayerClient() mongo.Client {
+	if container.mongoReplayerClient == nil {
+		var options = []mongo.Option{
+			mongo.WithLogger(container.GetLogger()),
+		}
+
+		container.mongoReplayerClient = mongo.NewReplayer(options...)
+	}
+
+	return container.mongoReplayerClient
+}
+
+func (container *Container) GetMongoWatcherClient() mongo.Client {
+	if container.mongoWatcherClient == nil {
+		var configOptions = container.Cfg.MongoDB.Options
+		var options = []mongo.Option{
+			mongo.WithLogger(container.GetLogger()),
+		}
+
+		container.mongoWatcherClient = mongo.NewWatcher(options...).WithOptions(
+			mongo.WithBatchSize(configOptions.BatchSize),
+			mongo.WithFullDocument(configOptions.FullDocument),
+			mongo.WithMaxAwaitTime(configOptions.MaxAwaitTime),
+		)
+	}
+
+	return container.mongoWatcherClient
+}
+
+func (container *Container) GetMongoConnection(ctx context.Context) *mongodriver.Database {
+	if container.mongoDB == nil {
+		mongoCfg := container.Cfg.MongoDB
+
+		if db, err := newMongoClient(ctx, container.GetLogger(), mongoCfg.URI, mongoCfg.DatabaseName); err != nil {
+			panic(err)
+		} else {
+			container.mongoDB = db
+		}
+	}
+
+	return container.mongoDB
+}
+
+func (container *Container) GetMongoCollection(ctx context.Context) mongo.CollectionAdapter {
+	if container.mongoCollection == nil {
+		container.mongoCollection = mongo.NewCollectionAdapter(container.GetMongoConnection(ctx).Collection(container.Cfg.MongoDB.CollectionName))
+	}
+
+	return container.mongoCollection
+}
+
+func newMongoClient(ctx context.Context, log logger.LoggerInterface, uri, database string) (*mongodriver.Database, error) {
+	opts := options.Client().ApplyURI(uri).SetReadPreference(readpref.Primary()).SetServerSelectionTimeout(2 * time.Second)
```

Comment:
Add option AppName on the client.
This is the name of the app expose by the driver to the server (and it can be retrieve in the server log files for analysis purpose).

Reply:
Thanks, I added the application name option :)